ipykernel/kernelspec.py

```diff
@@ -59,27 +59,37 @@ def get_kernel_dict(extra_arguments=None):
     }
 
 
-def write_kernel_spec(path=None, overrides=None, extra_arguments=None):
+def write_kernel_spec(path=None, overrides=None, extra_arguments=None, resources=RESOURCES):
```

Comment:
I'm ambivalent about this - on the one hand, I don't like adding function parameters just for tests, but on the other, I don't like using mocking unless it's necessary. So leave this as is unless someone else asks for a change.

Reply:
(Just realised 'as is' is ambiguous - I mean as you have already done it in this PR)

Reply:
Woo now I know how to use the mock patch thing :D never used it before.

Ran into a small problem though. I want to test that the mock patch worked correctly, and that the path the patched `RESOURCES` variable points to has the correct permissions set.

But the `RESOURCES` variable comes from `from ipykernel.kernelspec import RESOURCES` at the top, and I can't think of an easy way to get the patched variable without doing `import ipykernel.kernelspec` so that I can call `ipykernel.kernelspec.RESOURCES` directly in the `with mock.patch(...)` block.

At the same time I also think that this permission-changing behaviour should be a bit more verbose and raise a warning. So I did the whole "two birds with one stone" thing and added a warning to `write_kernel_spec` whenever the permissions are modified, and then the test checks that this warning is raised.

To be honest, I'm like 50/50 split on if the warning is useful or pointless, so if you think the warning shouldn't be there I'm happy to remove it.

Reply:
I think it shouldn't warn - read-only files in site-packages doesn't indicate that anything is going wrong or needs to be fixed. And people don't like unnecessary warnings.

If you want to ensure that it's picking up your test copy of the resources dir, why not stick an extra file in there, and then check that it exists in the resulting directory?

Reply:
Aha, good idea, implemented it like you said with a marker file.

My concern that `RESOURCES` didn't get mocked properly is probably silly, my thinking is more that the way the source resources directory is specified for `write_kernel_spec` might change at some point and just silently break this test.